iam/x509-identity-mgmt/js/src/services/CertificateService.js

```diff
@@ -32,17 +33,22 @@ class CertificateService {
    *
    * @returns an object containing the certificate in PEM format and its fingerprint.
    */
-  async generateCertificate({ csr: csrPem, belongsTo }) {
+  async generateCertificate({ csr: csrPem, belongsTo = {} }) {
     const csr = this.pkiUtils.parseCSR(csrPem);
 
     if (this.checkPublicKey) {
       this.pkiUtils.checkPublicKey(csr.subjectPublicKeyInfo);
     }
 
-    const subjectDN = this.dnUtils.from(csr.subject, true)
-      .verify()
-      .cnamePrefix(this.tenant)
-      .stringify();
+    const distinguishedNames = this.dnUtils.from(csr.subject, true).verify();
+
+    // ensure that the current tenant owns the device
+    await this.ensureOwner(distinguishedNames.CN);
```

Comment:
I thought about applying the validation for the CN while the IoTAgent MQTT is not enabled to work only with de fingerprint of the certificates. After the IotAgent is enabled, we can remove from x509-identity-mgmt all these validations that we do today in the CN field.

Reply:
I think we could set this behavior by configuration. This brings us the possibility of running a customized version of dojot without the device-manager. It would be nice to have this possibility.

Reply:
I liked that idea! We can think of this improvement as a kind of [Feature Toogle](https://en.wikipedia.org/wiki/Feature_toggle), but we would need to map the points that could be turned off and still the service remains healthy. :eyes: